haystack/pipelines/standard_pipelines.py

```diff
@@ -114,10 +115,82 @@ def eval(self,
         :param labels: The labels to evaluate on
         :param params: Params for the `retriever` and `reader`. For instance,
                        params={"Retriever": {"top_k": 10}, "Reader": {"top_k": 5}}
+        :param sas_model_name_or_path: SentenceTransformers semantic textual similarity model to be used for sas value calculation, 
+                                    should be path or string pointing to downloadable models.
         """
-        output = self.pipeline.eval(queries=queries, labels=labels, params=params)
+        output = self.pipeline.eval(queries=queries, labels=labels, params=params, 
+            sas_model_name_or_path=sas_model_name_or_path)
         return output
 
+    def _format_document_answer(self, document_or_answer: dict):
+        return "\n \t".join([f'{name}: {value}' for name, value in document_or_answer.items()])
+
+    def _format_retriever_query(self, query: dict):
+        metrics = "\n \t".join([f'{name}: {value}' for name, value in query['metrics'].items()])
+        documents = "\n\n \t".join([self._format_document_answer(doc) for doc in query['documents']])
+        gold_document_ids = "\n \t".join(query['gold_document_ids'])
+        s = (
+            f"Query: \n \t{query['query']}\n"
+            f"Gold Document Ids: \n \t{gold_document_ids}\n"
+            f"Metrics: \n \t{metrics}\n"
+            f"Documents: \n \t{documents}\n"
+            f"_________________________________________________"
+        )
+        return s
+
+    def _format_reader_query(self, query: dict):
+        metrics = "\n \t".join([f'{name}: {value}' for name, value in query['metrics'].items()])
+        answers = "\n\n \t".join([self._format_document_answer(answer) for answer in query['answers']])
+        gold_answers = "\n \t".join(query['gold_answers'])
+        s = (
+            f"Query: \n \t{query['query']}\n"
+            f"Gold Answers: \n \t{gold_answers}\n"
+            f"Metrics: \n \t{metrics}\n"
+            f"Answers: \n \t{answers}\n"
+            f"_________________________________________________"
+        )
+        return s
+
+    def print_eval_report(self, eval_result: EvaluationResult, n_worst_queries: int = 3):
```

Comment:
I find the final user flow a bit confusing:
1. call pipe.eval()
2. interact with the eval_results objects for everything that I want on top (metrics, worst queries)
3. go back to pipe.print_eval_report() if I want a report

I Intuitively looked for "print_eval_report()" in the EvalResults object. I suspect you moved it here to make clear that it only works with this retriever-reader-pipeline. However, even a manually initialized retriever-reader pipeline (via `pipe.add()` ) or a loaded one would not be able to use this method now. Feels to restrictive to me. 

Would it be an option to move this to `EvalResult` and restrict it there to certain pipeline types? Or moving it to the general pipeline class? Just thinking loud :)

Reply:
TL;DR:
It gets at least much more complicated if we move it to `EvalResult`. So I decided to stick it to the pipeline, as this is the only place that easily knows how a report for itself should look like. The basic idea here is, that `EvalResult` contains all potentially relevant metrics for each node in a very abstract way, so it's very general. But only the pipeline itself knows which node metrics are actually relevant for its evaluation.
 
Initially I tried to add `print_eval_report()` to `EvalResult`. However then some basic or even more than basic knowledge about the pipeline is required in EvalResult: we need to know at least which nodes follow other nodes, so we can draw a graph. Also there are some node specific metrics that make more sense to show in a report of a specific pipeline than others. My hypothesis is, that this is something only the pipeline itself knows best. Passing the whole pipeline into `EvalResult` seems not to be a good idea, as we wouldn't be able to load the EvalResult from disk without loading the whole pipeline again (and what if some parts of it like persisted document stores are not there anymore? and what about memory? EvalResult would always keep all pipeline components in memory). What seemed to be a bit better is to pass it the pipeline config that is used to save the pipeline to yaml. From there we could get the dependencies but it's also not really trivial and we would have to decide somewhere which metrics are relevant to evaluate the pipeline, too.

Reply:
I totally agree, that it's not super useful if we can only support `ExtractiveQA` pipelines. So I made some changes that I haven't yet commited to support all pipelines without junctions. And from this it shouldn't be too far to shift the `print_eval_report()` all over to `EvalResult` as suggested. Now I see 3 options:
- keep this PR as is and do the support for other pipelines and move of print_eval_report in another PR
- commit the changes to support all pipelines without junctions and do move of print_eval_report in another PR
- include the support for other pipelines and move of print_eval_report in this PR

I would go for option 3 as it should take at max two days to get there. However the PR grows and grows... 
@tholor, @julian-risch  what do you think?